src/App/App.jsx

```diff
@@ -0,0 +1,31 @@
+import { useState, createContext } from "react";
+import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
+import Navbar from "../components/Navbar";
+import MainSection from "../components/MainSection/MainSection";
+import CountryPage from "../components/CountryPage";
+import { Wrapper } from "./App.styles";
+
+export const isDarkModeContext = createContext();
+
+function App() {
+  const [isDarkMode, setIsDarkMode] = useState(false);
```

Comment:
często też jest tak, że tworzy się context jako osobny plik (tutaj np DarkMoceContext.jsx) i tam zawiera się caą logikę odpowiedzialną za zarządzanie trybem ciemnym, coś jak:
```
import React, { createContext, useState } from "react";

export const DarkModeContext = createContext({
  // default values: (to samo co jest w 'value' w providerze)
  isDarkMode: false,
  toggleIsDarkMode: () => {}
});

export const DarkModeContextProvider = ({ children }) => {
  const [isDarkMode, setIsDarkMode] = useState(false);
  const toggleIsDarkMode = setIsDarkMode((prev) => !prev);

  const value = {
    isDarkMode,
    toggleIsDarkMode
  };

  return (
    <DarkModeContext.Provider value={value}>
      {children}
    </DarkModeContext.Provider>
  );
};
```

Reply:
i wtedy podpinasz to tak:
```
<DarkModeContextProvider>
  <Router>
  .........
  </Router>
</DarkModeContextProvider>
```

Reply:
a używasz normalnie useContext(DarkModeContext)

Reply:
Jest to przydatne gdy context ma większą i bardziej złożoną logikę - wtedy wydziela się ją i nie brudzimy komponentu :D